internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1166,6 +1167,24 @@ func (r *BMACReconciler) ensureMCSCert(ctx context.Context, log logrus.FieldLogg
 		bmh.ObjectMeta.Annotations[BMH_AGENT_IGNITION_CONFIG_OVERRIDES] = ignitionWithMCSCert
 	}
 	log.Info("MCS certificate injected")
+
+	clusterInstall := hiveext.AgentClusterInstall{}
```

Comment:
@vrutkovs looking on it and i think i was wrong regarding the place. Sorry for it.
I think all this logic should move to clusterdeployment controller and be part of handleClusterInstalled

Reply:
@filanov what do you think?

Reply:
I think logic is right, just place is wrong

Reply:
This seems off.
This logic belongs in the agent not in bmac

Reply:
Once we leave `ensureMCSCert` there is no way to find out if IgnitionOverride is a custom one or created by `ensureMCSCert` (we of course could parse it and check `security` field). Not sure if moving parts to `handleClusterInstaller` is a good idea tbh

Reply:
Ah, wait, we can set custom Ignition URL in ACI initializer. That'd also solve
>Every BMH is going to update the ACI ignition endpoint?

Reply:
I think it was decided to always use https in ztp , @romfreiman ? so maybe it doesn't even depend on setting certificate at all? Just use https always?

Reply:
Moved this to clusterdeployment's `TransformClusterToDay2`.

Previously we always used insecure 22624, as on console.redhat.com we can't fetch MCS certificate and reliably store it for a long time